conan/internal/api/config/config_installer.py

```diff
@@ -127,13 +128,28 @@ def _process_file(directory, filename, config, cache_folder, folder):
         output.info("Copying file %s to %s" % (filename, target_folder))
         _filecopy(directory, filename, target_folder)
 
+def _find_conanignore(folder):
+    """ Traverse from the given folder up to the root folder looking for a .conanignore file."""
+    conanignore_path = os.path.join(folder, '.conanignore')
+    while not os.path.exists(conanignore_path):
+        parent_folder = os.path.dirname(folder)
+        if parent_folder == folder:
+            break
+        folder = parent_folder
+        conanignore_path = os.path.join(folder, '.conanignore')
+    if os.path.exists(conanignore_path):
+        conanignore_path = os.path.abspath(conanignore_path)
+    else:
+        conanignore_path = None
+    return conanignore_path
+
 
 def _process_folder(config, folder, cache_folder, ignore=None):
     if not os.path.isdir(folder):
         raise ConanException("No such directory: '%s'" % str(folder))
     if config.source_folder:
         folder = os.path.join(folder, config.source_folder)
-    conanignore_path = os.path.join(folder, '.conanignore')
+    conanignore_path = _find_conanignore(folder)
```

Comment:
Probably we just want to return here if ``None``, avoid checks above inside the matcher, and we can assume that if not None it will always be an abspath

Reply:
Not sure about this, maybe we just want to always read it from the root folder and not the source folder?  The truth is that this is not clearly documented about the expected location of this file, and this means it could break if assuming always in the root, so the traverse up seems more robust. Still, if some user was installing from a source folder, and their root .conanignore was not being processed, there is a small chance it might break someone if for example they are forcing the installation of a folder that would otherwise be ignored by a root install by the .conanignore. So it should be declared as a bug to move this forward.

Reply:
> Probably we just want to return here if None

Even if there's no conanignore, we sill need to run the for after this to process and install each file, no? The `matches` call will be a noop as the sets are empty

Reply:
Oh, yes, you are right, I was reading the ``for`` loop too quickly.

Then it might be just the above ``conanignore_path = os.path.abspath(conanignore_path)`` that was bothering me a bit, like it should either be ``None`` or an absolute path at that point.